source/common/upstream/eds.cc

```diff
@@ -214,9 +214,9 @@ bool EdsClusterImpl::updateHostsPerLocality(
   // out of the locality scheduler, we discover their new weights. We don't currently have a shared
   // object for locality weights that we can update here, we should add something like this to
   // improve performance and scalability of locality weight updates.
-  if (host_set.overprovisioningFactor() != overprovisioning_factor ||
-      updateDynamicHostList(new_hosts, *current_hosts_copy, hosts_added, hosts_removed,
+  if (updateDynamicHostList(new_hosts, *current_hosts_copy, hosts_added, hosts_removed,
```

Comment:
This is a really subtle bug. To prevent this from accidentally happening again can you potentially lift the call to `updateDynamicHostList` out of this if statement and assign it to its own const bool with a description of why it's done that way?